project/BuildSettings.scala

```diff
@@ -47,7 +47,7 @@ object BuildSettings {
     "-Ybackend-parallelism:16", // https://github.com/scala/scala3/pull/15392
     // "-nowarn", // during migration
     // "-rewrite",
-    // "-source:3.4-migration",
+    "-source:3.6-migration",
```

Comment:
This is to avoid false positive warning: https://github.com/scala/scala3/issues/21036, and force me to fix implicit ambiguity issue.